File: python/src/n0876_middle_of_the_linked_list_test.py
from n0876_middle_of_the_linked_list import middle_node
from utils.linked_list import List


def test_middle_node() -> None:
    linked_list = List([1, 2, 3, 4, 5])
    mid = middle_node(linked_list.head)
    assert mid is not None
    assert mid.value == 3

    linked_list = List([1, 2, 3, 4, 5, 6])
    mid = middle_node(linked_list.head)
    assert mid is not None
    assert mid.value == 4

File: rust/src/n0876_middle_of_the_linked_list.test.rs
use crate::n0876_middle_of_the_linked_list::middle_node;
use crate::utils::linked_list::List;

#[test]
fn test_middle_node() {
    let list = List::new([1, 2, 3, 4, 5]);
    let result = middle_node(list.head);
    assert_eq!(result.unwrap().value, 3);

    let list = List::new([1, 2, 3, 4, 5, 6]);
    let result = middle_node(list.head);
    assert_eq!(result.unwrap().value, 4);
}
